app/src/main/java/com/example/pokemonapp/fragment/CardTypeFragment.kt

```diff
@@ -5,14 +5,54 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.GridLayoutManager
+import com.example.pokemonapp.adapter.TypeAdapter
 import com.example.pokemonapp.databinding.FragmentCardTypeBinding
+import com.example.pokemonapp.utils.Event
+import com.example.pokemonapp.viewmodel.CardTypeViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class CardTypeFragment : Fragment() {
 
+    private lateinit var adapter: TypeAdapter
     private lateinit var binding: FragmentCardTypeBinding
+    private val viewModel by viewModel<CardTypeViewModel>()
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
         binding = FragmentCardTypeBinding.inflate(inflater, container, false)
+        viewModel.getLiveDataState().observe(viewLifecycleOwner, updateUIObserver)
```

Comment:
I think this methods should be in the on view created not when the view is create

Reply:
done